src/category_theory/limits/shapes/binary_products.lean

```diff
@@ -157,6 +157,30 @@ def has_binary_coproducts_of_has_colimit_pair [Π {X Y : C}, has_colimit (pair X
   has_binary_coproducts.{v} C :=
 { has_colimits_of_shape := { has_colimit := λ F, has_colimit_of_iso (diagram_iso_pair F) } }
 
+lemma prod.hom_ext [has_binary_products.{v} C] {Y A B : C} {a b : Y ⟶ A ⨯ B}
```

Comment:
```suggestion
@[ext] lemma prod.hom_ext [has_binary_products.{v} C] {Y A B : C} {a b : Y ⟶ A ⨯ B}
```

Not sure if this is a good idea.

Reply:
Would this conflict with `limit.hom_ext`, which is also marked with `[ext]`?

Reply:
Whichever one is defined last wins, which is just what we want in this situation.